tether/src/main/kotlin/com/fjordnet/tether/action/Action.kt

```diff
@@ -49,32 +53,38 @@ To use this with RxJava we had to create a class that contains a nullable type.
 This is because RxJava assumes everything to be non-null but LiveData can hold null values. I hate that we're going
 to have to check against this but until LiveData and RxJava play nice together we have to.
  */
-class Action<Input, Output> constructor(
-        private val execute: ((Input) -> Observable<Output>)?
-) : LiveData<Optional<Output>>(), LifecycleObserver {
+class Action<Input, Output> constructor(private val execute: ((Input) -> Observable<Output>)?) :
+    MutableLiveData<Optional<Output>>(), LifecycleObserver {
 
     /**
      * This emits whenever the underlying observable has completed.
      */
-    val completed: LiveData<Optional<Unit>> by lazy {
+    val completed: Observable<Unit> by lazy {
         _completed.value = Optional()
         _completed
+            .toForeverObservable()
+            .skipNull()
```

Comment:
For my own understanding - I thought Observable needed the same `Optional` as LiveData. Is this circumvented by calling `.skipNull`? Or am I misunderstanding this change?

Reply:
I added the skipNull() so whoever is subscribing to the executing, error, and completed Observables doesn't have to check if it is a valid emission by seeing if the nullable value the Optional is wrapping is null.